Sources/TokamakCore/Shapes/ShapeStyles/Gradient.swift

```diff
@@ -0,0 +1,440 @@
+  internal var gradient: Gradient
+  internal var center: UnitPoint
+  internal var startRadius: CGFloat
+  internal var endRadius: CGFloat
+
+  public init(gradient: Gradient, center: UnitPoint, startRadius: CGFloat, endRadius: CGFloat) {
+    self.gradient = gradient
+    self.center = center
+    self.startRadius = startRadius
+    self.endRadius = endRadius
+  }
+
+  public init(colors: [Color], center: UnitPoint, startRadius: CGFloat, endRadius: CGFloat) {
+    self.init(
+      gradient: Gradient(colors: colors), center: center,
+      startRadius: startRadius, endRadius: endRadius
+    )
+  }
+
+  public init(stops: [Gradient.Stop], center: UnitPoint, startRadius: CGFloat, endRadius: CGFloat) {
+    self.init(
+      gradient: Gradient(stops: stops), center: center,
+      startRadius: startRadius, endRadius: endRadius
+    )
+  }
+
+  public typealias Body = _ShapeView<Rectangle, Self>
+
+  public func _apply(to shape: inout _ShapeStyle_Shape) {
+    shape.result = .resolved(
+      .gradient(
+        gradient,
+        style: .radial(center: center, startRadius: startRadius, endRadius: endRadius)
+      )
+    )
+  }
+
+  public static func _apply(to type: inout _ShapeStyle_ShapeType) {}
+}
+
+@frozen public struct EllipticalGradient: ShapeStyle, View {
+  internal var gradient: Gradient
+  internal var center: UnitPoint
+  internal var startRadiusFraction: CGFloat
+  internal var endRadiusFraction: CGFloat
+
+  public init(
+    gradient: Gradient,
+    center: UnitPoint = .center,
+    startRadiusFraction: CGFloat = 0,
+    endRadiusFraction: CGFloat = 0.5
+  ) {
+    self.gradient = gradient
+    self.center = center
+    self.startRadiusFraction = startRadiusFraction
+    self.endRadiusFraction = endRadiusFraction
+  }
+
+  public init(
+    colors: [Color],
+    center: UnitPoint = .center,
+    startRadiusFraction: CGFloat = 0,
+    endRadiusFraction: CGFloat = 0.5
+  ) {
+    self.init(
+      gradient: .init(colors: colors),
+      center: center,
+      startRadiusFraction: startRadiusFraction,
+      endRadiusFraction: endRadiusFraction
+    )
+  }
+
+  public init(
+    stops: [Gradient.Stop],
+    center: UnitPoint = .center,
+    startRadiusFraction: CGFloat = 0,
+    endRadiusFraction: CGFloat = 0.5
+  ) {
+    self.init(
+      gradient: .init(stops: stops),
+      center: center,
+      startRadiusFraction: startRadiusFraction,
+      endRadiusFraction: endRadiusFraction
+    )
+  }
+
+  public typealias Body = _ShapeView<Rectangle, Self>
+
+  public func _apply(to shape: inout _ShapeStyle_Shape) {
+    shape.result = .resolved(
+      .gradient(
+        gradient,
+        style: .elliptical(
+          center: center,
+          startRadiusFraction: startRadiusFraction,
+          endRadiusFraction: endRadiusFraction
+        )
+      )
+    )
+  }
+
+  public static func _apply(to type: inout _ShapeStyle_ShapeType) {}
+}
+
+@frozen public struct AngularGradient: ShapeStyle, View {
```

Comment:
Could this be split into separate `AngularGradient.swift` file to fix the linter warning? May be applicable to other types in this file as well.